rajawali/src/main/java/org/rajawali3d/Object3D.java

```diff
@@ -174,30 +175,50 @@ protected void preRender() {
 	 * @param vpMatrix {@link Matrix4} The view-projection matrix
 	 * @param projMatrix {@link Matrix4} The projection matrix
 	 * @param vMatrix {@link Matrix4} The view matrix
-	 * @param pickerInfo The current color picker info. This is only used when an object is touched.
+	 * @param sceneMaterial The scene-wide Material to use, if any.
 	 */
 	public void render(Camera camera, final Matrix4 vpMatrix, final Matrix4 projMatrix,
 			final Matrix4 vMatrix, Material sceneMaterial) {
-		render(camera, vpMatrix, projMatrix, vMatrix, null, sceneMaterial);
-	}
+		render(camera, vpMatrix, projMatrix, vMatrix, null, sceneMaterial, false);
+	}
+
+    /**
+     * Renders the object with a parent matrix
+     *
+     * @param camera The camera
+     * @param vpMatrix {@link Matrix4} The view-projection matrix
+     * @param projMatrix {@link Matrix4} The projection matrix
+     * @param vMatrix {@link Matrix4} The view matrix
+     * @param parentMatrix {@link Matrix4} This object's parent matrix
+     * @param sceneMaterial The scene-wide Material to use, if any.
+     */
+    public void render(Camera camera, final Matrix4 vpMatrix, final Matrix4 projMatrix, final Matrix4 vMatrix,
+                       final Matrix4 parentMatrix, Material sceneMaterial) {
+        render(camera, vpMatrix, projMatrix, vMatrix, parentMatrix, sceneMaterial, false);
+    }
 
-	/**
-	 * Renders the object
-	 *
-	 * @param camera The camera
-	 * @param vpMatrix {@link Matrix4} The view-projection matrix
-	 * @param projMatrix {@link Matrix4} The projection matrix
-	 * @param vMatrix {@link Matrix4} The view matrix
-	 * @param parentMatrix {@link Matrix4} This object's parent matrix
-	 * @param pickerInfo The current color picker info. This is only used when an object is touched.
-	 */
+    /**
+     * Renders the object with a parent matrix, and optional color picking
+     *
+     * @param camera The camera
+     * @param vpMatrix {@link Matrix4} The view-projection matrix
+     * @param projMatrix {@link Matrix4} The projection matrix
+     * @param vMatrix {@link Matrix4} The view matrix
+     * @param parentMatrix {@link Matrix4} This object's parent matrix
+     * @param sceneMaterial The scene-wide Material to use, if any.
+     * @param isColorPicking True for color-picking render
+     */
 	public void render(Camera camera, final Matrix4 vpMatrix, final Matrix4 projMatrix, final Matrix4 vMatrix,
-			final Matrix4 parentMatrix, Material sceneMaterial) {
+			final Matrix4 parentMatrix, Material sceneMaterial, boolean isColorPicking) {
+
 		if (!mIsVisible && !mRenderChildrenAsBatch)
 			return;
 
 		Material material = sceneMaterial == null ? mMaterial : sceneMaterial;
-		preRender();
+
+        if (!isColorPicking) {
```

Comment:
I wasn't (and am still not) sure about the intended uses for `preRender()`; I didn't (at the time) find any usages in the library, so it seemed like a client hook, so I was concerned about having some way of indicating the rendering pass/context under which it was being called (object picking vs visual render passes) in case it involved a lot of processing or memory, but didn't want to modify its signature to pass along some kind of indicator.

Validating VBOs certainly sounds worthwhile, but maybe a task that does that could be queued to the GL thread as part of addChild() so it doesn't have to be checked on every render pass? just brainstorming. Any which way, I defer completely to your opinion here.